lightning/src/ln/peer_handler.rs

```diff
@@ -459,6 +459,16 @@ impl<Descriptor: SocketDescriptor, CM: Deref, L: Deref> PeerManager<Descriptor,
 		}
 	}
 
+	/// Append a message to a peer's pending outbound/write buffer, and update the map of peers needing sends accordingly
+	fn enqueue_message<M: Encode + Writeable>(&self, peers_needing_send: &mut HashSet<Descriptor>, peer: &mut Peer, descriptor: Descriptor, message: &M) {
```

Comment:
Not necessary to address now, but it would be worth thinking about the various ways we are sending messages in this module. Currently:
- log, enqueue, mark as needing send (here; called by `do_read_event`)
- log, enqueue (macro in `do_attempt_write_data`)
- enqueue (`process_events` and `timer_tick_occured`)

Reply:
`do_attempt_write_data differs` from do_read_event in that it doesn't update the peers needing send map, but it's also a very tiny method where the sending is self-contained. I am open for future brainstorming on its refactor. In principle, all sends should look the same anyway, and having the same method/macro name do different things would have been confusing, too.
At its core, `enqueue` is comprised of three components: serialization, the enqueueing, and the signaling mechanism. The signaling mechanism doubles the number of necessary arguments. Perhaps splitting that method into two, where one calls the other, would be appropriate?

Reply:
It's possible that `peers_needing_send` is just a historical artifact and thus no longer needed. See discussion in #456. So there may be some refactoring that will get us to one simple method.

The method could also be on `Peer` rather than `PeerManager`, though I think `Peer` would then need a logger...